### PR TITLE
Token authentication

### PR DESCRIPTION
**Pull request by ghost**
Fixes #

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by ghost** [comment]
for tickets:
https://github.com/muccg/bpa-archive-ops/issues/539
https://github.com/muccg/bpa-archive-ops/issues/541

**Comment by ghost** [comment]
docker-bpa-ckan - `next_release`
dockercompose-bpa-ckan - `incorporate_bpaotu` (BPAOTU and CKAN containers linked together)
ckan - `token_authentication`
bpaotu - `token_authentication`

Set:
BPAOTU_AUTH_SECRET_KEY = ""

File: ckan/controllers/user.py [hunk 1]
```diff
@@ -156,6 +158,28 @@ def me(self, locale=None):
         user_ref = c.userobj.get_reference_preferred_for_uri()
         h.redirect_to(locale=locale, controller='user', action='dashboard')
 
+    def check_permissions(self):
+        organisations = []
+
```
Comment (grahame) [inline ckan/controllers/user.py]: if the user isn't logged in we should just 403 this endpoint

File: ckan/config/routing.py [hunk 2]
```diff
@@ -390,6 +390,7 @@ def make_map():
         m.connect('/user/logged_out_redirect', action='logged_out_page')
         m.connect('/user/reset', action='request_reset')
         m.connect('/user/me', action='me')
+        m.connect('/user/check_permissions', action='check_permissions')
```
Comment (grahame) [inline ckan/config/routing.py]: could you make this URL clearly private? Maybe something like `/private/api/bpa/user_permissions`

File: ckan/controllers/user.py [hunk 3]
```diff
@@ -156,6 +158,28 @@ def me(self, locale=None):
...
+        user_organisations = h.organizations_available(permission='read')
+        for uo in user_organisations:
+            organisations.append(uo['name'])
+
+        # Note: the Ausmicro org name on prod is: 'australian-microbiome'
+        data_portion = {
+            'timestamp': time.time(),
+            'organisations': organisations
+        }
+
```
Comment (grahame) [inline ckan/controllers/user.py]: could you add in the user's registered email as well?

**Review by grahame — CHANGES_REQUESTED** [review]
Looks good, thanks Sam - just some minor changes please.

File: ckan/controllers/user.py (hunk 1 above)
Comment (ghost) [reply ckan/controllers/user.py]: Fixed.

File: ckan/controllers/user.py (hunk 3 above)
Comment (ghost) [reply ckan/controllers/user.py]: Fixed.

File: ckan/config/routing.py (hunk 2 above)
Comment (ghost) [reply ckan/config/routing.py]: Fixed.

**Review by grahame — APPROVED** [review]

**Pull request merged**